### PR TITLE
Various typo fixes

### PR DESCRIPTION
**Pull request by rouault**
Including 3 public methods, which are deprecated by this commit and replaced by a typo-fixed version

### CODE REVIEW

File: python/PyQt6/server/auto_generated/qgsserverogcapi.sip.in [hunk 1]
```diff
@@ -132,9 +132,19 @@ Returns the string representation of a ``ct`` (Content-Type) attribute.
 Returns the file extension for a ``ct`` (Content-Type).
 %End
 
-    static QgsServerOgcApi::ContentType contenTypeFromExtension( const std::string &extension );
+ static QgsServerOgcApi::ContenType contenTypeFromExtension( const std::string &extension ) /Deprecated/;
```
Comment (nyalldawson) [inline python/PyQt6/server/auto_generated/qgsserverogcapi.sip.in]:
```suggestion
 static QgsServerOgcApi::ContentType contenTypeFromExtension( const std::string &extension ) /Deprecated/;
```

File: src/gui/symbology/qgssvgselectorwidget.h [hunk 2]
```diff
@@ -406,8 +406,15 @@ class GUI_EXPORT QgsSvgSelectorWidget : public QWidget, private Ui::WidgetSvgSel
     /**
      * Returns if the group box to fill parameters is visible
      * \since QGIS 3.18
+     * \deprecated Use allowParameters()
      */
-    bool allowParamerters() const {return mAllowParameters;}
+    Q_DECL_DEPRECATED bool allowParamerters() const {return mAllowParameters;} SIP_DEPRECATED // spellok
```
Comment (nyalldawson) [inline src/gui/symbology/qgssvgselectorwidget.h:411]:
```suggestion
    Q_DECL_DEPRECATED bool allowParamerters() const SIP_DEPRECATED {return mAllowParameters;} // spellok
```

Comment (rouault) [reply src/gui/symbology/qgssvgselectorwidget.h]: done

**Review by nyalldawson — APPROVED** [review]

**Comment by rouault** [comment]
I'm confused by the clang-tidy warnings (https://github.com/qgis/QGIS/actions/runs/8073755030/job/22064545754?pr=56567) which seem unrelated to my changes. Can we ignore them without breaking future builds / pull requests?

**Comment by nyalldawson** [comment]
@rouault 

It runs on touched files only, so any issues in those files get flagged regardless of whether the pr introduced them. It's safe to ignore existing issues from this test.

**Pull request merged**